allennlp/interpret/attack/hotflip.py

```diff
@@ -0,0 +1,184 @@
+            all_tokens_tokenized = self.predictor._dataset_reader._token_indexers["token_characters"]._character_tokenizer.batch_tokenize(all_tokens)                        
+            pad_length = max([len(x) for x in all_tokens_tokenized])    
+            character_tokens = []            
+            if getattr(all_tokens_tokenized[0][0], 'text_id', None) is not None:                
+                for tok in all_tokens_tokenized:
+                    tmp = [x.text_id for x in tok]                    
+                    tmp += [0] * (pad_length-len(tmp))
+                    character_tokens.append(tmp)                            
+            else:
+                for tok in all_tokens_tokenized:
+                    tmp = \
+                        [self.vocab.get_token_index(x.text, \
+                        self.predictor._dataset_reader._token_indexers["token_characters"]._namespace) for x in tok]                    
+                    tmp += [0] * (pad_length-len(tmp))
+                    character_tokens.append(tmp)                
+            character_tokens = torch.LongTensor(character_tokens)
+            all_inputs["token_characters"] = character_tokens.unsqueeze(0)
+
+            if "elmo" in self.predictor._dataset_reader._token_indexers:
+                pad_length = pad_length + 2 # elmo has start/end word
+                elmo_tokens = []                
+                for tok in all_tokens:                    
+                    tmp = self.predictor._dataset_reader._token_indexers["elmo"].tokens_to_indices([Token(text=tok)], self.vocab,"sentence")["sentence"]                                    
+                    elmo_tokens.append(tmp[0])                
+                all_inputs["elmo"] = torch.LongTensor(elmo_tokens).unsqueeze(0)                 
+
+        # find the TextFieldEmbedder
+        for module in self.predictor._model.modules():
+            if isinstance(module, TextFieldEmbedder):
+                model = module  
+        # pass all tokens through the fake matrix and create an embedding out of it.
+        embedding_matrix = model(all_inputs).squeeze()                        
+        self.token_embedding = Embedding(num_embeddings=self.vocab.get_vocab_size('tokens'), embedding_dim=embedding_matrix.shape[1], weight=embedding_matrix, trainable=False)
+
+    def attack_from_json(self, inputs:JsonDict, target_field: str, gradient_index:str):        
+        """ 
+        Replaces one token at a time from the input until the model's prediction changes.
+        """ 
+        og_instances = self.predictor.inputs_to_labeled_instances(inputs)        
+        original = list(og_instances[0][target_field].tokens)        
+        final_tokens = []        
+        for i in range(len(og_instances)):            
+            new_instances = [og_instances[i]]                        
+
+            # handling fields that need to be checked            
```

Comment:
I don't know what this means.  I'm not going to try to figure out what this logic does right now; I'll wait until you've broken it up and given some more comments to look at this method again.